src/run.ts

```diff
@@ -52,6 +56,24 @@ export async function pyRunApp(): Promise<void> {
     },
   });
 
+  // Wait until the server port and auto-reload ports are both available.
+  const serverPortsAvailable = await Promise.all([
+    waitUntilServerPortIsAvailable(port),
+    waitUntilServerPortIsAvailable(autoreloadPort),
+  ]);
+
+  if (!serverPortsAvailable.every((x) => x)) {
+    if (!serverPortsAvailable[0]) {
+      vscode.window.showErrorMessage(`Unable to open server port ${port}.`);
+    }
+    if (!serverPortsAvailable[1]) {
```

Comment:
Maybe `else if` so you don't get both messages

Reply:
Oh, good point. I think I can simplify the logic too then.